### PR TITLE
Blog: Apache DataFusion is now the fastest single node engine for querying Apache Parquet files

### PR DESCRIPTION
**Pull request by alamb**
- Closes https://github.com/apache/datafusion/issues/13436

Let's celebrate the accomplishment of getting to the top of the ClickBench leaderboard

Screenshot 2024-11-18 at 10 38 27 AM

### CODE REVIEW

File: _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md
Comment (Omega359) [inline _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md:103]:
```suggestion
out of [single 2x performance improvements] several years ago. Working together our
```

File: _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md
Comment (Omega359) [inline _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md:118]:
```suggestion
[ClickHouse](https://clickhouse.com/) which were not designed to be extended.
```

File: _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md
Comment (Omega359) [inline _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md:164]:
```suggestion
Parquet with Millisecond Latency] )
```

File: _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md
Comment (pmcgleenon) [inline _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md:162]:
```suggestion
Part of the reason for DataFusion's speed in ClickBench is reading Parquet files (really) quickly,
```

File: _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md
Comment (pmcgleenon) [inline _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md:183]: **Skipping Partial Aggregation When Doesn't Help**

Is there a missing word in this title?

Comment (alamb) [reply _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md:183]:
```suggestion
## Skipping Partial Aggregation When It Doesn't Help
```

**Comment by comphead** [comment]
Thats greatest news. Congrats!

**Comment by comphead** [comment]
For the ClickBench run for DataFusion what is the `batch_size` being used? is it 8192 by default?

**Comment by alamb** [comment]
> For the ClickBench run for DataFusion what is the `batch_size` being used? is it 8192 by default?

I don't think the scripts change the default setting -- the scripts used are here: https://github.com/ClickHouse/ClickBench/tree/main/datafusion

Here is the PR to update for 43.0.0: https://github.com/ClickHouse/ClickBench/pull/251

File: _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md
Comment (Weijun-H) [inline _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md:156]:
```suggestion
* **DataFusion**:  [Xiangpeng Hao](https://github.com/XiangpengHao), again charted the initial path and [Weijun Huang](https://github.com/Weijun-H), [Dharan Aditya](https://github.com/dharanad) [Lordworms](https://github.com/Lordworms), [Jax Liu](https://github.com/goldmedal),  [wiedld](https://github.com/wiedld), [Tai Le Manh](https://github.com/tlm365), [yi wang](https://github.com/my-vegetable-has-exploded), [doupache](https://github.com/doupache), [Jay Zhan](https://github.com/jayzhan211) , [Xin Li](https://github.com/xinlifoobar)  and [Kaifeng Zheng](https://github.com/Kev1n8) made it real.
```

File: _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md
Comment (jayzhan211) [inline _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md:264]:
```suggestion
Zhan](https://github.com/jayzhan211) tried [several], [different] approaches until arriving
```

File: _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md
Comment (pmcgleenon) [inline _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md:203]:
```suggestion
For such queries, the first aggregation phase does not significantly
```

File: _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md
Comment (tlm365) [inline _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md:88]:
```suggestion
[Mehmet Ozan Kabak]: https://www.linkedin.com/in/mehmet-ozan-kabak/
```

**Review by tlm365 — CHANGES_REQUESTED** [review]

**Review by pmcgleenon — APPROVED** [review]

**Comment by alamb** [comment]
I would like to merge this PR to publish it tomorrow if there are no more comments. I can't so so without another committer clicking the approve button:

Screenshot 2024-11-20 at 9 58 21 AM

File: _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md
Comment (alamb) [reply _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md:83]: I certainly was inspired by the comment to help me focus where I spent my time reviewing PRs and helping push them through, though it is a good point that this may imply it motivated others as well, when I don't really know what did. 

Perhaps we could rephrase the motivation with something like this?

"Performance has long been a focus for DataFusion: one of the core benefits of DataFusion is its core performance, which  both excites contributors and attracts users. There seems to have been a renewed focus on performance recently, including a call in July 2024 from Mehmet ....?

Comment (Dandandan) [reply _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md:83]: Sounds good to me

Comment (alamb) [reply _posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md]: I rephrased in 2507e67

**Review by jayzhan211 — APPROVED** [review]

**Review by viirya — APPROVED** [review]

**Review by Weijun-H — APPROVED** [review]

**Comment by alamb** [comment]
Thank you to everyone who reviewed this PR. I plan to merge / publish it later today unless there are any other comments

**Comment by adriangb** [comment]
Amazing work Andrew! You and all of the DataFusion contributors should be incredibly proud of this accomplishment.

**Comment by alamb** [comment]
Let's get this published to the world

**Comment by alamb** [comment]
https://datafusion.apache.org/blog/2024/11/18/datafusion-fastest-single-node-parquet-clickbench/

**Pull request merged**
